registered-comuni-pn/src/services/read-model-queries.service.ts

```diff
@@ -0,0 +1,104 @@
+import { MongoClient } from 'mongodb'
+import { env } from '../configs/env.js'
+import { Tenant, tenantSchema, purposeSchema, Purpose } from '../models/index.js'
+import {
+  COMUNI_E_LORO_CONSORZI_E_ASSOCIAZIONI_ATTRIBUTE_ID,
+  PN_ESERVICE_ID,
+} from '../configs/constants.js'
+
+export class ReadModelQueriesClient {
+  private client: MongoClient
+
+  private constructor(client: MongoClient) {
+    this.client = client
+  }
+
+  /**
+   * Connects to the mongodb database
+   */
+  public static async connect() {
+    const connectionConfig = {
+      replicaSet: 'rs0',
+      readPreference: 'secondaryPreferred',
+    } as const
+
+    // Use this config instead if you want to connect to a mongodb instance locally using a tunnel
+    // const connectionConfig = {
+    //   directConnection: true,
+    //   readPreference: 'secondaryPreferred',
+    //   retryWrites: false,
+    // } as const
+
+    const connectionString = `mongodb://${env.READ_MODEL_DB_USER}:${env.READ_MODEL_DB_PASSWORD}@${env.READ_MODEL_DB_HOST}:${env.READ_MODEL_DB_PORT}`
+    const client = await new MongoClient(connectionString, connectionConfig).connect()
+    return new ReadModelQueriesClient(client)
+  }
+
+  /**
+   * Retrieves all the purposes related to the e-service with id PN_ESERVICE_ID, having at
+   * least one version with state Active, Suspended or WaitingForApproval.
+   */
+  async getPNEServicePurposes() {
+    return await this.client
+      .db(env.READ_MODEL_DB_NAME)
+      .collection<{ data: Purpose }>(env.PURPOSES_COLLECTION_NAME)
+      .find(
+        {
+          'data.eserviceId': PN_ESERVICE_ID,
+          'data.versions': {
+            $elemMatch: {
+              state: { $in: ['Active', 'Suspended', 'WaitingForApproval'] },
+            },
+          },
```

Comment:
very minor: feel free to verify, but this could be simplified as
```suggestion
          'data.versions.state': { $in: ['Active', 'Suspended', 'WaitingForApproval'] }
```

Reply:
Thanks for the suggestion, gonna test it out

Reply:
Not working, the `$in` operator needs the data to be an array, and `data.versions.state` is not

Reply:
uhm this is weird, I just tried this query and it works
```
db.purposes.find({"data.versions.state": {$in: ['Active', 'Suspended', 'WaitingForApproval']}})
```
The `$in` operator should require that the set you are comparing to is an array, but the collection field should not